internal/config/config.go

```
@@ -41,21 +47,79 @@ type Config struct {
 	WebhookPort            int            `yaml:"webhookPort"`
 	Worker                 Worker         `yaml:"worker"`
 }
+type ConfigGetter interface {
+	GetConfig(ctx context.Context, logger logr.Logger, userConfigMapName, userConfigMapNamespace string) (*Config, error)
+}
+
+//go:generate mockgen -source=config.go -package=config -destination=mock_config.go
+type configHelperAPI interface {
+	getClient() (client.Client, error)
+	loadConfigFromCM(cm *corev1.ConfigMap, cfg *Config) error
+}
+type configGetter struct {
+	configHelper configHelperAPI
+}
+
+func NewConfigGetter() ConfigGetter {
+	return &configGetter{
+		configHelper: newConfigHelper(),
+	}
+}
+
+type configHelper struct {
+}
+
+func newConfigHelper() configHelperAPI {
+	return &configHelper{}
+}
 
-func ParseFile(path string) (*Config, error) {
-	fd, err := os.Open(path)
+func (ch *configHelper) getClient() (client.Client, error) {
+	clnt, err := client.New(ctrl.GetConfigOrDie(), client.Options{})
 	if err != nil {
-		return nil, fmt.Errorf("could not open the configuration file: %v", err)
+		return nil, fmt.Errorf("failed to create new client %v", err)
 	}
-	defer fd.Close()
+	return clnt, err
+}
+
+func (ch *configHelper) loadConfigFromCM(cm *corev1.ConfigMap, cfg *Config) error {
+	const configKey = "controller_config.yaml"
+	rawConfig, ok := cm.Data[configKey]
```

What happens if `cm == nil`? Can we return an error in case `cm == nil` ?